.github/workflows/main.yml

```diff
@@ -68,12 +68,12 @@ jobs:
      - name: Run unit-tests
        run: make unit-test
      - name: Upload codecoverage
-       uses: codecov/codecov-action@4fe8c5f003fae66aa5ebb77cfd3e7bfbbda0b6b0 # 3.1.5
+       uses: codecov/codecov-action@e0b68c6749509c5f83f984dd99a76a1c1a231044 # 4.0.1
        with:
          files: ./unit-coverage.out
          verbose: true
```

Comment:
seems like we'll need to deal with a token

Reply:
https://github.com/ossf/scorecard/actions/runs/7805848954/job/21290943573?pr=3857#step:10:53
> Error: Codecov token not found. Please provide Codecov token with -t flag.

Reply:
hmm, it's complicated by the fact that we run this in a `pull_request` trigger, which doesn't have access to secrets.